v2/spiffetls/tlsconfig/config.go

```diff
@@ -0,0 +1,103 @@
+package tlsconfig
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/x509bundle"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+)
+
+// TLSClientConfig returns a TLS configuration which verifies and authorizes
+// the server X509-SVID.
+func TLSClientConfig(svid x509svid.Source, bundle x509bundle.Source, authorizer x509svid.Authorizer) *tls.Config {
+	panic("not implemented")
+}
+
+// SetTLSClientConfig sets TLS configuration which verifies and authorizes
+// the server X509-SVID.
+func SetTLSClientConfig(config *tls.Config, svid x509svid.Source, bundle x509bundle.Source, authorizer x509svid.Authorizer) {
```

Comment:
@evan2645 and I talked IRL and decided that `Hook*` would be a more appropriate name for the indended behavior here. I'll update.